doc/1.7/dev_guide/lua_style_guide.rst

```diff
@@ -0,0 +1,465 @@
+      end -- good, but not explicit
+
+* Type conversion
+
+  Do not use concatenation to convert to string or addition to convert to number
+  (use ``tostring``/``tonumber`` instead):
+
+  .. code-block:: lua
+
+      local a = 123
+      a = a .. ''
+      -- bad
+
+      local a = 123
+      a = tostring(a)
+      -- good
+
+      local a = '123'
+      a = a + 5 -- 128
+      -- bad
+
+      local a = '123'
+      a = tonumber(a) + 5 -- 128
+      -- good
+
+* Try to avoid multiple nested if's or conditions with common body:
+
+  .. code-block:: lua
+
+      if (a == true and b == false) or (a == false and b == true) then
+          do_something()
+      end
+      -- good
+
+      if a == true then
+          if b == false then
+              do_something()
+          end
+      if b == true then
+          if a == false then
+              do_something()
+          end
+      end
+      -- bad
+
+* Avoid multiple concatenations in one parameter, use string.format instead:
+
+  .. code-block:: lua
+
+      function say_greeting(period, name)
+          local a = "good  " .. period .. ", " .. name
+      end
+      -- bad
+
+      function say_greeting(period, name)
+          local a = string.format("good %s, %s", period, name)
+      end
+      -- good
+
+      local say_greeting_fmt = "good %s, %s"
+      function say_greeting(period, name)
+          local a = say_greeting_fmt:format(period, name)
+      end
+      -- best
+
+* ``and``/``or`` for default variable values or..
+
+  .. code-block:: lua
+
+    function(input)
+        input = input or 'default_value'
+    end -- good
+
+    function(input)
+        if input == nil then
+            input = 'default_value'
+        end
+    end -- ok, but excessary
+
+* If's and simple cases:
+
+  .. code-block:: lua
+
+      if a == true then
+          return do_something()
+      end
+      do_other_thing() -- good
+
+      if a == true then
+          return do_something()
+      else
+          do_other_thing()
+      end -- bad
+
+* Using spaces:
+
+  - do not use a space after functions and after commas in arguments:
+
+    .. code-block:: lua
+
+        function name(arg1, arg2, ...)
+        end
+
+  - use spaces after comments:
+
+    .. code-block:: lua
+
+        while true do -- inline comment
+        -- comment
+        --[[ multiline
+        comment ]]--
+        do_something()
+        end
+
+  - surrounding operators:
+
+    .. code-block:: lua
+
+        local thing=1
+        thing = thing-1
+        thing = thing*1
+        thing = 'string'..'s'
+        -- bad
+
+        local thing = 1
+        thing = thing - 1
+        thing = thing * 1
+        thing = 'string' .. 's'
+        -- good
+
+  - use space after commas in tables:
+
+    .. code-block:: lua
+
+        local thing = {1,2,3}
+        thing = {1 , 2 , 3}
+        thing = {1 ,2 ,3}
+        -- bad
+
+        local thing = {1, 2, 3}
+        -- good
+
+  - Add a line break after multiline blocks.
```

Comment:
Maybe we should stated expicitly that it should be *one* blank line btw functions / blocks — for ones who aware of PEP-0008 :)